src/ensembl/src/content/app/species-selector/components/species-search-field/SpeciesSearchField.tsx

```diff
@@ -64,6 +65,11 @@ export const SpeciesSearchField = (props: Props) => {
 
     const speciesName = getSpeciesAnalyticsName(match);
 
+    analyticsTracking.setCustomDimension({
+      diemension: CustomDimensions.SPECIES,
+      value: match.genome_id
+    });
```

Comment:
to middleware?

Reply:
Created separate setter functions for dimensions as discussed.